src/components/list/listcontainer.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div class='uk-panel uk-width-1-1'>
+    <ul class='uk-list uk-list-divider'>
+      <list-item
+        v-for='item in items'
+        :key='item.uid'
+        :uid='item.uid'
+        :editor='EditorClass'
+        @delete-item='onDeleteItem'
+        @change-item='onChangeItem'
+      ></list-item>
+    </ul>
+    <div class='uk-panel uk-margin'>
+      <button class="uk-button uk-button-primary uk-float-right"
+      @click="createItem">Add Item</button>
+    </div>
+  </div>
+</template>
+
+<script>
+import Vue from 'vue'
+import ListItem from './listitem.vue'
+import ModalDialog from '../modaldialog.vue'
+
+export default {
+  name: 'list-container',
+  components: {
+    ListItem,
+  },
+  props: {
+    editor: {
+      type: Object,
+      required: true
+    },
+  },
+  data () {
+    return {
+      currentItem: {},
+      ModalDialogClass: Vue.extend(ModalDialog),
+      EditorClass: null,
+      dialogInstance: null,
+      items: {},
+    }
+  },
+  methods: {
+    createItem () {
+      this.currentItem = {}
+      this.dialogInstance = new this.ModalDialogClass({
+        propsData: {
+          component: this.editor,
+          properties: this.currentItem
+        }
+      })
+      this.dialogInstance.$on('cancel', this.onCancelNewItem)
+      this.dialogInstance.$on('save', this.onSaveNewItem)
+      this.dialogInstance.show()
+    },
+    onDeleteItem (key) {
+      let temp = Object.assign({}, this.items)
+      delete temp[key]
+      this.items = temp
+    },
+    onChangeItem (data) {
+      // this.$emit('list-changed', this.serialize())
+    },
+    onCancelNewItem () {
+      this.$set(this, 'dialogInstance', null)
+    },
+    onSaveNewItem (data) {
+      this.$store.dispatch('updateDataSource', data)
+      let newItem = new this.EditorClass({
+        propsData: {
+          uid: data.uid,
+          dataSourceType: data.type,
+          attributes: Object.assign({}, data.attributes),
+          icon: data.icon,
+          clusterPoints: data.clusterIcons,
+          polygonStyle: data.polygonStyle,
+          popover: data.popover,
+        }
+      })
+      let tempObj = {}
+      tempObj[newItem.uid] = newItem
+      this.items = Object.assign({}, this.items, tempObj)
+      this.$set(this, 'dialogInstance', null)
+      this.$emit('list-changed', this.serialize())
+    },
+    serialize () {
```

Comment:
Is `serialize` doing anything for you? My expectation is that serialization would be coming out of the store rather than the UI.

Reply:
The mapeditor currently doesn't have access to the store. Like I said previously, the store was added in late, and so we could do this in a refactor.